### PR TITLE
Rust kit

### LINKED ISSUE #111: Build Rust starter kit
Build a Rust starter kit for lux 2022.

I am learning to program Rust, and this has captivated my attention as a good project for learning.

### PR DESCRIPTION
**Pull request by Seamooo**
closes #111 

Remaining work:
- [x] Better path finding
- [x] Forward sim?
- [x] Errors over panics
- [x] Executable verification
- [x] Coverage tests
- [x] Integration tests

Clarifications:
- Are factories guaranteed to to have an origin at least `(1,1)`, and at most `(size-2, size-2)`?
- Are the player ids guaranteed to be `player_0`, `player_1` every match? If so
  is it worthwhile calling them a generic player_id or just denoting them the
  same way `LIGHT` and `HEAVY` is denoted?
- What's the use of factions / should names outside of the 4 valid values be supported?
- I've seen some implementations use BigInteger representations for bid amounts, is this necessary?

### CODE REVIEW

**Comment by Seamooo** [comment]
- Better path finding has been labelled as a nice to have once confirming that submitting an invalid action in the demo doesn't cause a crash
- Forward sim has not been implemented as it seems like effectively a no-op in the python implementation due to simulating actions not being possible
- Executable has been verified for Linux and Windows platforms
- Coverage tests have been written for code outside of serializing / deserializing and communication
- Integration tests take the form of the demo

**Comment by Seamooo** [comment]
@StoneT2000 is there anything additional required to get this merged?

**Comment by StoneT2000** [comment]
Thanks for the contribution @Seamooo I will try to review it this weekend and give feedback on what might need changing.

File: kits/rust/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+game rules.
+
+**Disclaimer**: Although this kit is heavily documented and  reasonably tested.
+It has been implemented by a third party based on the specification and python
+implementation. As such, any deviation from the python implementation with
+regards to documentation or behaviour is unintended and should be logged as a bug
+
+## Implementation Clarifications
+
+- What's the use of factions / should names outside of the 4 valid values be supported
```
Comment (StoneT2000) [inline kits/rust/README.md:14]: Factions are currently unused, but there is only those 4 values. Feel free to remove this section.

**Review by StoneT2000 — CHANGES_REQUESTED** [review]

**Review by StoneT2000 — APPROVED** [review]

**Comment by StoneT2000** [comment]
thank you for the contribution! @Seamooo

**Pull request merged**